File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:

        ans, banned = -1, set(banned)

        for i in range(1,n+1):

            if i in banned: continue 
            
            maxSum-= i
            ans += 1
            if maxSum < 0: return ans

        return ans+1

File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        Set<Integer> set = new HashSet<>();
        for (int i : banned) {
            set.add(i);
        }

        int count = 0;
        for (int i = 1; i <= n; i++) {
            if (set.contains(i)) {
                continue;
            }
            if (i <= maxSum) {
                count++;
                maxSum -= i;
            } else {
                break;
            }
        }

        return count;
    }
}